src/App.tsx

```diff
@@ -9,36 +9,109 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { getUsers, getPosts } from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [currentUserPosts, setCurrentUserPosts] = useState<Post[]>([]);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [activePost, setActivePost] = useState<null | Post>(null);
+
+  const getUserPosts = async (user: User) => {
+    setIsError(false);
+    setIsLoadingPosts(true);
+
+    try {
+      const response = await getPosts(user.id);
+
+      setCurrentUserPosts(response);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoadingPosts(false);
+    }
+  };
+
+  const hideCommemts = () => {
+    setActivePost(null);
+    setCurrentUserPosts([]);
+  };
+
+  const handleUserSelect = (user: User) => {
+    setCurrentUser(user);
+  };
+
+  const handleSelectActivePost = (post: Post) => {
+    setActivePost(post);
+  };
+
+  useEffect(() => {
+    getUsers()
+      .then(value => {
+        setUsers(value);
+      });
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  currentUser={currentUser}
+                  onUserSelect={handleUserSelect}
+                  getUserPosts={getUserPosts}
+                  hideCommemts={hideCommemts}
```

Comment:
again, instead of passing these 2 functions better execute this logic here, in the `handleUserSelect` function